src/main/java/com/unzer/payment/service/UrlUtil.java

```diff
@@ -36,151 +37,187 @@
-		if (stringBuilder.charAt(stringBuilder.length()-1) != '/') {
-			stringBuilder.append("/");
-		}
-	}
-
-	public String getHirePurchaseRateUrl(BigDecimal amount, Currency currency, BigDecimal effectiveInterestRate, Date orderDate) {
-		StringBuilder stringBuilder = new StringBuilder();
-		stringBuilder.append(getRestUrl());
-		stringBuilder.append("types/hire-purchase-direct-debit/plans?");
-		stringBuilder.append("amount=").append(getBigDecimal(amount)).append("&");
-		stringBuilder.append("currency=").append(currency.getCurrencyCode()).append("&");
-		stringBuilder.append("effectiveInterest=").append(getBigDecimal(effectiveInterestRate)).append("&");
-		stringBuilder.append("orderDate=").append(getDate(orderDate));
-		return stringBuilder.toString();
-	}
-
-	private String getDate(Date date) {
-		return new SimpleDateFormat("yyyy-MM-dd").format(date);
-	}
-
-	private String getBigDecimal(BigDecimal decimal) {
-		NumberFormat df = NumberFormat.getNumberInstance(Locale.ENGLISH);
-		df.setMaximumFractionDigits(4);
-		df.setMinimumFractionDigits(0);
-		df.setGroupingUsed(false);
-		return df.format(decimal);
-	}
+        return stringBuilder.toString();
+    }
+
+    public String getHttpGetUrl(String id) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(getRestUrlWithOutPaymentType());
+        appendSlashIfNeeded(stringBuilder);
+        stringBuilder.append(id);
+        return stringBuilder.toString();
+    }
+
+    public String getRecurringUrl(Recurring recurring) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(getRestUrl());
+        appendSlashIfNeeded(stringBuilder);
+        stringBuilder.append(RECURRING_URL);
+        String result = stringBuilder.toString();
+        result = result.replace(PLACEHOLDER_TYPE_ID, recurring.getTypeId());
+        return result;
+    }
+
+    public String getRestUrl(PaymentType paymentType) {
+        return getRestUrlInternal(paymentType).replace("<paymentId>/", "");
+    }
+
+    public String getRestUrlWithOutPaymentType() {
+        return getRestUrlInternal().replace("<paymentId>/", "");
+    }
+
+    @Deprecated
+    private String getRestUrlInternal(Basket basket) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(properties.getString(PropertiesUtil.REST_ENDPOINT));
+        appendSlashIfNeeded(stringBuilder);
+
+        if (basket.isV1()) {
+            stringBuilder.append("v1");
+        } else if (basket.isV2()) {
+            stringBuilder.append("v2");
+        } else {
+            stringBuilder.append("v1");
+            appendSlashIfNeeded(stringBuilder);
+            stringBuilder.append("migration");
+        }
+        appendSlashIfNeeded(stringBuilder);
+        stringBuilder.append(basket.getTypeUrl());
+        return stringBuilder.toString();
+    }
+
+    private String getRestUrlInternal(PaymentType paymentType) {
```

Comment:
Payment type and Basket should be two completely different Classes.
An object of type PaymentType can never be an instance of Basket.